File: database-driver-plugin/src/main/java/de/lino/database/database/sql/mysql/MySQLDatabaseProvider.java
package de.lino.database.database.sql.mysql;

import de.lino.database.database.auth.Credentials;
import de.lino.database.database.DatabaseType;
import de.lino.database.database.sql.SQLDatabaseProvider;
import de.lino.database.database.sql.SQLExecution;
import lombok.Getter;
import org.jetbrains.annotations.NotNull;

/**
 * The {@link SQLDatabaseProvider} for MySQL, connected via {@link SQLExecution} using
 * {@link DatabaseType#MY_SQL}'s JDBC driver and URL scheme.
 */
@Getter
public class MySQLDatabaseProvider extends SQLDatabaseProvider {

    /**
     * Connects to a MySQL database with {@code credentials}.
     *
     * @param credentials the login credentials and connection details to connect with
     */
    public MySQLDatabaseProvider(@NotNull Credentials credentials) {
        super(DatabaseType.MY_SQL, new SQLExecution(DatabaseType.MY_SQL, credentials));
    }

}

File: python/database-driver-plugin/src/database_driver/plugin/database/sql/mysql/mysql_database_provider.py
"""Mirror of ``de.lino.database.database.sql.mysql.MySQLDatabaseProvider``."""

from __future__ import annotations

from database_driver.api.database.auth.credentials import Credentials
from database_driver.api.database.database_type import DatabaseType
from database_driver.plugin.database.sql.sql_database_provider import SQLDatabaseProvider
from database_driver.plugin.database.sql.sql_execution import SQLExecution


class MySQLDatabaseProvider(SQLDatabaseProvider):
    """The ``SQLDatabaseProvider`` for MySQL, connected via ``SQLExecution`` using
    ``DatabaseType.MY_SQL``'s driver and connection scheme."""

    def __init__(self, credentials: Credentials) -> None:
        super().__init__(DatabaseType.MY_SQL, SQLExecution(DatabaseType.MY_SQL, credentials))
